test/extensions/upstreams/tcp/generic/config_test.cc

```diff
@@ -0,0 +1,38 @@
+#include "extensions/upstreams/tcp/generic/config.h"
+
+#include "test/mocks/tcp/mocks.h"
+#include "test/mocks/upstream/cluster_manager.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::NiceMock;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace Upstreams {
+namespace Tcp {
+namespace Generic {
+
+class TcpConnPoolTest : public ::testing::Test {
+public:
+  NiceMock<Upstream::MockThreadLocalCluster> thread_local_cluster_;
+  GenericConnPoolFactory factory_;
+  NiceMock<Envoy::Tcp::ConnectionPool::MockUpstreamCallbacks> callbacks_;
+};
+
+TEST_F(TcpConnPoolTest, TestNoHost) {
+  envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy_TunnelingConfig config;
```

Comment:
Arguably there is a host, just no connection to it, but that's super nitty so feel free to merge in as is :-)